trlx/pipeline/offline_pipeline.py

```diff
@@ -103,9 +103,17 @@ class PromptPipeline(BasePipeline):
     Tokenizes prompts, unless they are already tokenized, and truncates them to `max_prompt_length` from the right
     """
 
-    def __init__(self, prompts: List[str], max_prompt_length: int, tokenizer: PreTrainedTokenizer):
+    def __init__(
+        self, prompts: Union[Dict[str, Any], List[str]], max_prompt_length: int, tokenizer: PreTrainedTokenizer
+    ):
         super().__init__()
 
+        if isinstance(prompts[0], dict):
+            metadata = prompts
+            prompts = [x.pop("prompt") for x in metadata]
```

Comment:
In `PromptPipeline`'s constructor docstring, we should document that `"prompt"` is an expected field of `metadata` when `prompts` consists of dictionaries.